hybridauth.module

```diff
@@ -12,8 +12,235 @@ function hybridauth_form_alter(&$form, FormStateInterface $form_state, $form_id)
+  $providers = hybridauth_providers_list();
+  return isset($providers[$provider_id]) ? $providers[$provider_id] : NULL;
+}
+
+/**
+ * Internal functions.
+ */
+function hybridauth_providers_list($reset = FALSE) {
+  $providers = &drupal_static(__FUNCTION__, NULL);
+
+  if (!isset($providers)) {
+    $raw_providers = [
+      // Core providers.
+      'AOL' => t('AOL'),
+      'Facebook' => t('Facebook'),
+      'Foursquare' => t('Foursquare'),
+      'Google' => t('Google'),
+      'LinkedIn' => t('LinkedIn'),
+      'Live' => t('Windows Live'),
+      'MySpace' => t('MySpace'),
+      'OpenID' => t('OpenID'),
+      'Twitter' => t('Twitter'),
+      'Yahoo' => t('Yahoo'),
+      // Additional providers.
+      'px500' => t('500px'),
+      'Disqus' => t('Disqus'),
+      'FamilySearch' => t('FamilySearch'),
+      'Geni' => t('Geni'),
+      'GitHub' => t('GitHub'),
+      'Goodreads' => t('Goodreads'),
+      'HumanitarianId' => t('Humanitarian ID'),
+      'Identica' => t('Identica'),
+      'Instagram' => t('Instagram'),
+      'LastFM' => t('LastFM'),
+      'Mailru' => t('Mail.ru'),
+      'Murmur' => t('Murmur'),
+      'MyHeritage' => t('MyHeritage'),
+      'Odnoklassniki' => t('Odnoklassniki'),
+      'Pixnet' => t('Pixnet'),
+      'Plurk' => t('Plurk'),
+      'QQ' => t('QQ'),
+      'Sina' => t('Sina'),
+      'Skyrock' => t('Skyrock'),
+      'Steam' => t('Steam'),
+      'Tumblr' => t('Tumblr'),
+      'TwitchTV' => t('Twitch.tv'),
+      'Viadeo' => t('Viadeo'),
+      'Vimeo' => t('Vimeo'),
+      'Vkontakte' => t('VKontakte'),
+      'Yandex' => t('Yandex'),
+    ];
+    foreach (hybridauth_providers_files($reset) as $name => $file) {
+      if (!array_key_exists($name, $raw_providers)) {
+        $raw_providers[$name] = t($name);
+      }
+    }
+
+    $providers = array();
+    $enabled_providers = array_filter(\Drupal::state()->get('hybridauth_providers', []));
+    foreach (array_keys($enabled_providers + $raw_providers) as $provider_id) {
+      $providers[$provider_id] = $raw_providers[$provider_id];
+    }
   }
 
+  return $providers;
 }
+
+/**
+ * Returns available providers files, keyed by filename without extension.
+ */
+function hybridauth_providers_files($reset = FALSE) {
+  $files = array();
+  // Try to get cached value if allowed: $reset == FALSE.
+  if (!$reset) {
+    $cache = \Drupal::cache()->get('hybridauth_providers_files');
+    if (!empty($cache->data)) {
+      $files = $cache->data;
+    }
+  }
+
+  // Scan Providers directory if $reset == TRUE or there was no cached value.
+  if (empty($files)) {
+    $files = \Drupal::service('file_system')
+      ->scanDirectory(
+        _hybridauth_library_path() . '/hybridauth/src/Provider',
+        '/\.php$/',
+        ['key' => 'name']
+      );
+    \Drupal::cache()->set('hybridauth_providers_files', $files);
+  }
+  return $files;
+}
+
+/**
+ * Returns the path to the HybridAuth library.
+ */
+function _hybridauth_library_path() {
```

Comment:
this is poor port of d7 - I don't think we need to have it at all since we are now with full Composer support.

Reply:
This function is deleted. Also added code for getting the location of the vendor directory.